dlt/destinations/impl/duckdb/configuration.py

```diff
@@ -25,6 +25,7 @@ class DuckDbBaseCredentials(ConnectionStringCredentials):
     read_only: bool = False  # open database read/write
 
     def borrow_conn(self, read_only: bool) -> Any:
+        # TODO: Can this be done in sql client instead?
```

Comment:
no - in loader each thread gets a private sql client and connection. however in duckdb there's just a single connection that can be borrowed. So I need a "singleton" which the instance of configuration is.

I agree that we could implement such singleton separately but this is close to perfect - each configuration describes a particular duckdb database so this mechanism fits here pretty well...

Reply:
Yeah I get it. I ran into a problem trying to copy the config why I added the comment, but it's not a big deal right now I think.   
I agree it's pretty neat like this and nice to not need another global weakref mapping or something.